python/ray/data/_internal/planner/plan_read_op.py

```diff
@@ -36,6 +38,38 @@ def cleaned_metadata(read_task):
     return block_meta
 
 
+def _splitrange(n, k):
+    """Calculates array lens of np.array_split().
+
+    This is the equivalent of
+    `[len(x) for x in np.array_split(range(n), k)]`.
+    """
+    base = n // k
+    output = [base] * k
+    rem = n - sum(output)
+    for i in range(len(output)):
+        if rem > 0:
+            output[i] += 1
+            rem -= 1
+    assert rem == 0, (rem, output, n, k)
+    assert sum(output) == n, (output, n, k)
+    return output
```

Comment:
why not just do `[len(x] for x in np.array_split(range(n), k)`?

Reply:
This PR only moves this function from another file. I don't know if there was a particular reason. cc @ericl